File: apps/desktop/src/api/hardware.ts
import { api } from "./client";
import type { HardwareProfileResponse, HardwareSnapshot } from "./types";

export const hardwareApi = {
  scan: () => api.post<HardwareProfileResponse>("/hardware/scan"),
  latest: () => api.get<HardwareProfileResponse | null>("/hardware/latest"),
  live: () => api.get<HardwareSnapshot>("/hardware/live"),
};

File: core/potato_core/api/hardware.py
from __future__ import annotations

from fastapi import APIRouter, HTTPException

from potato_core.engines.hardware import service
from potato_core.schemas.hardware import HardwareProfileResponse, HardwareSnapshot

router = APIRouter(prefix="/hardware", tags=["hardware"])


@router.post("/scan", response_model=HardwareProfileResponse)
def scan() -> HardwareProfileResponse:
    """Full detection pass. Called on first launch and on-demand re-scan."""
    return service.scan_and_persist()


@router.get("/latest", response_model=HardwareProfileResponse | None)
def latest() -> HardwareProfileResponse | None:
    return service.get_latest_profile()


@router.get("/live", response_model=HardwareSnapshot)
def live() -> HardwareSnapshot:
    """Cheap poll for the live hardware monitor widget."""
    return service.poll_live()
